### ISSUE TITLE
[mcs] Tuple destruction gives incorrect output

### ISSUE THREAD

**Issue by chenzhekl** [mono/mono#15735]
## Steps to Reproduce

1. Not sure if this is a bug or not. Run the following code with Mono 5.20.1.34

```
using System;

namespace a
{
    class Program
    {
        static void Main(string[] args)
        {
            int a = 10;
            int b = 6;
            (a, b) = (b, a);
            Console.WriteLine($"{a}, {b}");
        }
    }
}
```

### Current Behavior

Outputs `6, 6`

### Expected Behavior

Outputs `6, 10` as in DotNet Core 2.2

## On which platforms did you notice this

[ ] macOS
[x] Linux
[ ] Windows

**Version Used**: 5.20.1.34

## Stacktrace

```
Please paste the stack trace here if available.
```

**Comment by lambdageek** [comment]
Looks like an `mcs` problem.  If I compile the example with `csc`, I get the expected `6,10` output.

**Comment by lewurm** [comment]
Was it fixed by this https://github.com/mono/mono/pull/12395 ? Merged June 5th.

**Comment by lambdageek** [comment]
Yes, this is fixed on master.

Broken: `6.4.0.81 (2019-06/f55e0a671c0 Mon Jul  1 13:17:57 EDT 2019)`.
Fixed: `6.5.0.582 (master/d5b374bb51a Wed Jul 17 19:18:29 EDT 2019)`

**Comment by lewurm** [comment]
Okay, sounds like it's fixed then. The recommendation is anyway to use Roslyn (`csc`). Thanks for your report @chenzhekl!

**Comment by lewurm** [comment]
Just saw that @lambdageek triggered a backport to our current release branch: https://github.com/mono/mono/pull/15747

So this fix will land in Mono 6.4 (aka. `2019-06`).
